reducer/src/test/java/com/graphicsfuzz/reducer/reductionopportunities/VariableDeclToExprReductionOpportunitiesTest.java

```diff
@@ -0,0 +1,152 @@
+    assertTrue(ops.isEmpty());
+  }
+
+  @Ignore
+  @Test
+  public void testMultipleDeclarations() throws Exception {
+    final String program = "void main() {"
+        + "int a = 1;"
+        + "int b = foo();"
+        + "int c;"
+        + "}";
+    final String expected = "void main() {"
+        + " int a;"
+        + " a = 1;"
+        + " int b;"
+        + " b = foo();"
+        + " int c;"
+        + "}";
+    final TranslationUnit tu = ParseHelper.parse(program);
+    List<VariableDeclToExprReductionOpportunity> ops = VariableDeclToExprReductionOpportunities
+        .findOpportunities(MakeShaderJobFromFragmentShader.make(tu), new ReducerContext(true,
+            ShadingLanguageVersion.ESSL_100,
+            new RandomWrapper(0), null, true));
+    assertEquals(2, ops.size());
+    for (int i = 0; i < ops.size(); i++) {
+      ops.get(i).applyReduction();
+    }
+    CompareAsts.assertEqualAsts(expected, tu);
+  }
+
+  @Ignore
+  @Test
+  public void testMultipleLineDeclarationsOneLine() throws Exception {
+    final String program = "void main() {"
+        + "int a;"
+        + "int b = 1, c, d = foo(), e, f = bar(); "
+        + "int g;"
+        + "}";
+    final String expected = "void main() {"
+        + " int a;"
+        + " int b, c, d, e, f;"
+        + " b = 1;"
+        + " d = foo();"
+        + " f = bar();"
+        + " int g;"
+        + "}";
+    final TranslationUnit tu = ParseHelper.parse(program);
+    List<VariableDeclToExprReductionOpportunity> ops = VariableDeclToExprReductionOpportunities
+        .findOpportunities(MakeShaderJobFromFragmentShader.make(tu), new ReducerContext(true,
+            ShadingLanguageVersion.ESSL_100,
+            new RandomWrapper(0), null, true));
+    assertEquals(3, ops.size());
+    for (int i = 0; i < ops.size(); i++) {
+      ops.get(i).applyReduction();
+    }
+    CompareAsts.assertEqualAsts(expected, tu);
+  }
+
+  @Ignore
+  @Test
+  public void testAssignVariableIdentifier() throws Exception {
+    final String program = "void main() {"
+        + "int a = 1;"
+        + "int b = a, c = b;"
+        + "int d = c;"
+        + "}";
+    final String expected = "void main() {"
+        + " int a;"
+        + " a = 1;"
+        + " int b, c;"
+        + " b = a;"
+        + " c = b;"
+        + " int d;"
+        + " d = c;"
+        + "}";
+    final TranslationUnit tu = ParseHelper.parse(program);
+    List<VariableDeclToExprReductionOpportunity> ops = VariableDeclToExprReductionOpportunities
+        .findOpportunities(MakeShaderJobFromFragmentShader.make(tu), new ReducerContext(true,
+            ShadingLanguageVersion.ESSL_100,
+            new RandomWrapper(0), null, true));
+    assertEquals(4, ops.size());
+    for (int i = 0; i < ops.size(); i++) {
+      ops.get(i).applyReduction();
+    }
+    CompareAsts.assertEqualAsts(expected, tu);
+  }
+
+  @Ignore
+  @Test
+  public void testReplaceGlobalDeclaration() throws Exception {
```

Comment:
I believe the following test is invalid, as it is illegal to have an assignment statement at global scope.

I like the idea of having a transformation that removes global initializers, but it would have to move them into main (as the first statements in main), and I think that would be better as a separate reducer pass.  So please remove this test and file an issue for us to add such a pass.

Reply:
Fixed